### PR TITLE
This adds better syntax error handling

### PR DESCRIPTION
**Pull request by bbakerman**
Today the parser will throw a ANTLR specific exception and then it gets interpreted in two places.

Depending on the path take, the line number and column is not always included

Instead I wanted to make it more powerful with the line and column always captured and to get the offending token as well as a source preview around that line

This will produce an exception like

```
InvalidSyntaxError{ message=Invalid Syntax : 'Offending token is 'scala'' ,locations=[SourceLocation{line=2, column=12}] ,sourcePreview=
            scala Url   # spillin misteak

            interface Foo {
               is_foo : Boolean
            }
}
```

Which is easier to see what bit of syntax is wrong.

### CODE REVIEW

File: src/main/java/graphql/InvalidSyntaxError.java [hunk 1]
```diff
@@ -2,32 +2,37 @@
...
         this.message = mkMessage(msg);
+        this.sourcePreview = sourcePreview;
         if (sourceLocations != null) {
             this.locations.addAll(sourceLocations);
         }
     }
 
     private String mkMessage(String msg) {
-        return "Invalid Syntax" + (msg == null ? "" : " : " + msg);
+        return "Invalid Syntax" + (msg == null ? "" : " : " + msg + "");
```
Comment (felipe-gdr) [inline src/main/java/graphql/InvalidSyntaxError.java]: do we need that empty string at the end?

File: src/main/java/graphql/parser/ExtendedBailStrategy.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        String msg = null;
+        String sourcePreview = null;
+        Token currentToken = recognizer.getCurrentToken();
+        if (currentToken != null) {
+            line = currentToken.getLine();
+            column = currentToken.getCharPositionInLine();
+            sourcePreview = mkPreview(line);
+            msg = "Offending token is '" + currentToken.getText() + "'";
+        }
+        return new InvalidSyntaxError(singletonList(new SourceLocation(line, column, sourceName)), msg, sourcePreview, cause);
```
Comment (felipe-gdr) [inline src/main/java/graphql/parser/ExtendedBailStrategy.java]: can the resulting message end up having a `-1,-1` location if `currentToken` result in a `null` value?

Comment (bbakerman) [reply src/main/java/graphql/parser/ExtendedBailStrategy.java]: fixed

File: src/main/java/graphql/InvalidSyntaxError.java (hunk 1 above)
Comment (bbakerman) [reply src/main/java/graphql/InvalidSyntaxError.java]: fixed

**Pull request merged**